lib/std/math/big/int.zig

```diff
@@ -3885,7 +3887,7 @@ fn llsignedor(r: []Limb, a: []const Limb, a_positive: bool, b: []const Limb, b_p
         // x & ~a can only clear bits, so (x & ~a) <= x, meaning (-b - 1) + 1 never overflows.
         assert(r_carry == 0);
 
-        // With b = 0 and b_borrow = 0, we get ~a & (-0 - 0) = ~a & 0 = 0.
+        // With b = 0 and b_borrow = 0, we get ~a & (0 - 0) = ~a & 0 = 0.
```

Comment:
I don't see a point to removing the `-` in these.

Reply:
My understanding is that the `b` in the equation the second part of this comment is based on:
```
r = a | (--b)
  = a | ~(-b - 1)
  = ~~a | ~(-b - 1)
  = ~(~a & (-b - 1))
  = -((~a & (-b - 1)) + 1)
       |-----------|
```
is a negative number, but the `b` in the code is its absolute value.
And I think that the `b = 0` part of this comment is saying that the upper part of an absolute value is zero.

Reply:
But don't the limbs always have the absolute value since the sign is stored separately.

Reply:
@Vexu Yes, but the `a` and `b` in the equation `r = a + b = -((~a & (-b - 1)) + 1)` are meant to be *numbers*, not *absolute values*. In fact, we derive this equation exactly because the *number* `b` is negative in this branch and we want to have `-b` (i.e. its absolute value) in the equation since that's what we store.